Pocket/pocket-ios
# feat(premium): offline purchase alert

## Summary
* Show alert if user is offline and trying to purchase premium subscription

## References 
* [IN-1147](https://getpocket.atlassian.net/browse/IN-1147)

## Implementation Details
* added network analyzer to determine offline state in AccountViewModel

## Test Steps
* as a non premium user,
* go to settings, click on "Go Premium"
* premium upgrade sheet should appear
* go back to settings
* go offline
* click on "Go Premium"
* alert should appear
* clicking "OK" should dismiss alert
* clicking "Go to Settings" should take you to the app settings

## PR Checklist:

## Screenshots
The difference in UI is determined by OS, not something I can control. 

<img width="390" alt="image" src="https://user-images.githubusercontent.com/111293209/224783267-7d47aef9-9730-4754-8e02-6dfb54b84996.png">

<img width="394" alt="image" src="https://user-images.githubusercontent.com/111293209/224783315-500afdf1-8c83-4279-b18d-418a03c79007.png">


[IN-1147]: https://getpocket.atlassian.net/browse/IN-1147?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
